packages/backend-core/src/events/processors/async/DocumentUpdateProcessor.ts

```diff
@@ -0,0 +1,44 @@
+import { EventProcessor } from "../types"
+import { Event, Identity, DocUpdateEvent } from "@budibase/types"
+import { DocumentType, SEPARATOR } from "../../../constants"
+import { doInTenant } from "../../../context"
+import { getDocumentId } from "../../documentId"
+import { shutdown } from "../../asyncEvents"
+
+export type Processor = (update: DocUpdateEvent) => Promise<void>
```

Comment:
We could remove:
- `DocUpdateEvent`
- `Processor`
In favour of:
- `EventPayload`
- `EventProcessor` 


I don't see a lot of benefit in implementing a sub event type system when we could simplify and re-use the top level ones.

Reply:
This is the same as previous comment - I really prefer having the specific processors which aren't aware of the event system - in the actual usage you just want a simple, functional interface where you pass a function that happens when a document type is updated - completely separate from event processing (a somewhat complex and very un-related system).